src/schml/insert-casts.rkt

```diff
@@ -229,14 +237,17 @@
        ;; source type is not known at this point anyway. However, we
        ;; can detect injections in the coercions case to be optimized
        ;; later by some cast optimizer.
-       (match e-ty
+              (match e-ty
          [(Dyn)
           (let ([lbl (mk-label "munbox" e-src)])
             (cond
               [(dynamic-operations?)
-               (Dyn-MRef-Ref e (lbl))]
+               (define dop (Dyn-MRef-Ref e (lbl)))
```

Comment:
I thought about it, but I don't think the code would be any clearer, and it would be a modest gain in space savings.